### DISCUSSION TITLE
Ingesting syslogs from firewall with SO 2

### DISCUSSION THREAD

**Discussion by doritoes** [Security-Onion-Solutions/securityonion#2136] [Unsupported Versions] [answered]
How do I get my firewall syslogs into SO2 and parsed properly like I did previously?

Previously I used so-allow, then created two files, and pointed syslog to SO:
/etc/logstash/custom/1005_preprocess_syslog_types.conf
/etc/logstash/custom/6299_firewall_usg.conf

It was straightforward to bring in my Unifi USG (or palo alto or pfsense) firewall logs and map them to the right fields so the SO "firewall" dashboards worked great.

Is there a step-by-step documentation on getting firewall syslog logs working with SO2?

The ingest page is quite confusing as it implies that syslog ingest is only available in Eval mode.
https://docs.securityonion.net/en/2.3/ingest.html?highlight=ingest

Some statements seem contradictory: "Since Logstash no longer parses logs in Security Onion 2, modifying existing parsers or adding new parsers should be done via Elasticsearch." vs "All custom logstash pipeline configurations should be placed in /opt/so/saltstack/local/salt/logstash/pipelines/config/custom."
https://docs.securityonion.net/en/2.3/logstash.html#adding-new-logs-or-modifying-existing-parsing

In my humble opinion firewall logs (especially the IDS alerts) are a vital log source for the SIEM.

Thank you.

**Comment by dougburks** [comment, marked as answer]
I've updated the following pages:
https://docs.securityonion.net/en/2.3/logstash.html
https://docs.securityonion.net/en/2.3/syslog.html
https://docs.securityonion.net/en/2.3/ingest.html

We already have an Elasticsearch ingest parser for pfsense firewall logs sent via syslog which you can use as an example:
https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/filterlog

Hope that helps!

**Comment by mtnsec** [comment]
> How do I get my firewall syslogs into SO2 and parsed properly like I did previously?
> 
> Previously I used so-allow, then created two files, and pointed syslog to SO:
> /etc/logstash/custom/1005_preprocess_syslog_types.conf
> /etc/logstash/custom/6299_firewall_usg.conf
> 
> It was straightforward to bring in my Unifi USG (or palo alto or pfsense) firewall logs and map them to the right fields so the SO "firewall" dashboards worked great.

@doritoes would you be willing to share your conf files for the previous version of SO?  We are reluctant to move to SO2 yet based on initial testing, but have a need to be able to parse Unifi logs in the shorter term.

**Reply by doritoes** [reply]
@mtnsec sure. I broke down the old environment and am slowly working through 3 deployment scenarios with the new version and documenting as I go including wazuh, osquery, Anomali STAXX.

Here's what I saved and was working with Unifi. You need to set your IPs in there. The cloudkey bit is there, but I never got logs to test with. No edgerouter testing as I wasn't able to get a cheap edgerouter on ebay.

/etc/logstash/custom/1005_preprocess_syslog_types.conf
```
# Author: doritoes
# Last Update: 9/13/2020
# /etc/logstash/custom/1005_preprocess_syslog_types.conf
# Logstash copies to /etc/logstash/conf.d/1005_preprocess_syslog_types.conf
filter {
  # first USG, have tested with USG, not yet PRO
  if "syslog" in [tags] {
    if [host] == "gateway" {
      mutate {
        replace => { "type" => "usg" }
        add_tag => [ "firewall", "usg", "unifi" ]
      }
    }
    # next an access points, have tested with UAP-AC-LR
    if [host] == "192.168.10.18" {
      mutate {
        add_tag => [ "unifi", "unifi_syslog", "access_point" ]
        add_field => { "syslog-host_from" => [host] }
      }
    }
    # have not tested with a switch yet
    # finally the cloudkey, have not received any logs yet
    if [host] == "192.168.10.253" {
      mutate {
        add_tag => [ "unifi", "unifi-controller" ]
        add_field => { "syslog-host_from" => [host] }
      }
    }
    mutate {
        #add_tag => [ "conf_file_1005"]
    }
  }
}
```

/etc/logstash/custom/6299_firewall_usg.conf
```
# Author: doritoes
# Last Update: 9/13/2020
# /etc/logstash/custom/6299_firewall_usg.conf
# Logstash copies to /etc/logstash/conf.d/6299_firewall_usg.conf

filter {
  if [type] == "usg" {
    mutate {
      gsub => [ "message", "= ", "=NA " ]
      gsub => [ "message", " DF ", " DF=1 " ]
      gsub => [ "message", " ACK ", " ACK=1 " ]
      gsub => [ "message", " SYN ", " SYN=1 " ]
      gsub => [ "message", " PSH ", " PSH=1 " ]
    }
    grok {
       match => ["message", "^\[%{DATA:reason}\]%{GREEDYDATA:kv}"]
       tag_on_failure => ["usgnotfirewall"]
    }
    if "usgnotfirewall" in [tags] {
      mutate {
        add_tag => [ "unifi_syslog" ]
        remove_tag => [ "usgnotfirewall", "firewall" ]
        remove_field => [ "type" ]
      }
    }
# Unifi appends a "D" for deny to deny rules
# and "A" for allow
    if [reason] =~ /D$/ {
      mutate {
        add_field => { "action" => "deny" }
      }
    } else if [reason] =~ /A$/ {
      mutate {
        add_field => { "action" => "allow" }
      }
    } else {
      mutate {
        add_field => { "action" => "unknown" }
      }
    }
    kv {
      source => "kv"
    }
    mutate {
      rename => {  "SRC" => "source_ip" }
      rename => {  "DST" => "destination_ip" }
      rename => {  "MAC" => "source_mac" }
      rename => {  "IN" => "source_interface" }
      rename => {  "OUT" => "destination_interface" }
      rename => {  "SPT" => "source_port" }
      rename => {  "DPT" => "destination_port" }
      rename => {  "PROTO" => "protocol" }
      rename => {  "LEN" => "payload_length" }
      convert => [ "destination_port", "integer" ]
      convert => [ "source_port", "integer" ]
      remove_field => [ "kv", "log" ]
      replace => { "event_type" => "firewall" }
      replace => { "type" => "firewall" }
      remove_tag => [ "syslogng" ]
    }
    if "unifi_syslog" in [tags] {
      mutate {
        replace => { "event_type" => "syslog" }
        remove_field => [ "type", "action" ]
      }
    }
    if [addr_ip] {
      mutate {
        add_field => { "ips" => [ "%{addr_ip}" ] }
      }
    }
    if [assign_ip] {
      mutate {
        add_field => { "ips" => [ "%{assign_ip}" ] }
      }
    }
    if [assigned_ip] {
      mutate {
        add_field => { "ips" => [ "%{assigned_ip}" ] }
      }
    }
    if [date] and [time] {
      mutate {
        add_field => { "receive_time" => "%{date} %{time}" }
        remove_field => [ "date", "time" ]
      }
      date {
        timezone => "America/New_York"
        match => [ "receive_time", "YYYY-MM-dd HH:mm:ss" ]
        target => "receive_time"
      }
      mutate {
        rename => { "receive_time" => "@timestamp" }
      }
    } else {
      mutate {
        add_tag => [ "missing_date" ]
      }
    }
    mutate {
      #add_tag => [ "conf_file_6299"]
    }
  }
}
```

**Reply by mtnsec** [reply]
This is excellent, thank you!!!
